src/Service/Dist/Storage/FileStorage.php

```diff
@@ -42,7 +43,9 @@ public function download(string $url, Dist $dist, array $headers = []): void
 
         AtomicFile::write(
             $filename,
-            $this->downloader->getContents($url, $headers)->getOrElseThrow(
+            $this->downloader->getContents($url, $headers, function () use ($url): void {
+                throw new NotFoundHttpException(sprintf('File not found at %s', $url));
```

Comment:
this is quick and nasty fix, because this layer should not use this exception, but at the moment I don't see a better and faster way to fix this problem